Recrutify/Recrutify.Services/Services/ScheduleService.cs

```diff
@@ -81,9 +110,60 @@ public async Task UpdateScheduleSlotsForCurrentUserAsync(IEnumerable<DateTime> d
         }
 
+        private async Task<IEnumerable<Interview>> GetInterviews(IEnumerable<CandidateDTO> candidates, IEnumerable<InterviewAppointmentDTO> interviewAppointmentDTOs, Guid projectId)
+        {
+            var nameAndEmailUsers = await _userService.GetNamesAndEmailsByIdsAsync(interviewAppointmentDTOs.Where(i => i.IsAppointment == true).Select(i => i.UserId));
+            var interviews = new List<Interview>();
+            foreach (var candidate in candidates)
+            {
+                var projectResult = candidate.ProjectResults.FirstOrDefault(pr => pr.ProjectId == projectId);
+                var userId = interviewAppointmentDTOs.FirstOrDefault(i => i.CandidateId == candidate.Id).UserId;
+                if (userId != Guid.Empty)
+                {
+                    var interview = _mapper.Map<Interview>(candidate);
+                    interview.User = new UserEmailInfo()
+                    {
+                        Id = userId,
+                        Email = nameAndEmailUsers.TryGetValue(userId, out var strEmail) ? strEmail.Split(',')[0] : default,
+                        Name = nameAndEmailUsers.TryGetValue(userId, out var strName) ? strName.Split(',')[1] : default,
+                    };
+                    interview.AppointDateTimeUtc = interviewAppointmentDTOs.FirstOrDefault(i => i.CandidateId == candidate.Id).DateTime;
+                    interview.InterviewType = projectResult.Status == StatusDTO.Test ? InterviewType.RecruityInterview
+                        : projectResult.Status == StatusDTO.RecruiterInterview ? InterviewType.TechnicalInterviewOne
+                        : InterviewType.TechnicalInterviewSecond;
+                }
+            }
+
+            return interviews;
+        }
+
         private IEnumerable<DateTime> GetDateTimeInScheduleSlots(IEnumerable<ScheduleSlot> scheduleSlots)
         {
             return scheduleSlots?.Select(s => s.AvailableTime) ?? new List<DateTime>();
         }
+
+        private IEnumerable<ScheduleCandidateInfoDTO> GetScheduleCandidateInfos(IEnumerable<CandidateDTO> candidates, Guid projectId)
+        {
+            var scheduleCandidateInfos = new List<ScheduleCandidateInfoDTO>();
+            foreach (var candidate in candidates)
+            {
+                var projectResult = candidate.ProjectResults.FirstOrDefault(pr => pr.ProjectId == projectId);
+                if (!projectResult.IsAssignedOnInterview)
+                {
+                    var scheduleCandidateInfo = _mapper.Map<ScheduleCandidateInfoDTO>(candidate);
+                    scheduleCandidateInfo.ProjectResult = new ScheduleCandidateProjectResultDTO()
+                    {
+                        IsAssignedOnInterview = true,
+                        PrimarySkill = projectResult.PrimarySkill,
+                        ProjectId = projectResult.ProjectId,
+                        Status = projectResult.Status + 1,
```

Comment:
If user will try to assign after candidate is in tech 2 interview it'll be exception